coreos-ostree-importer/coreos_ostree_importer.py

```diff
@@ -184,23 +183,13 @@ def get_sha256sum(filepath: str) -> str:
     return h.hexdigest()
 
 
-def parse_s3_url(url: str) -> tuple:
-    if not url.startswith("s3://"):
-        raise Exception(f"Unable to parse the s3 url: {url}")
-    # Chop off s3:// and break into bucket / key
-    bucket, key = url[5:].split("/", 1)
-    return (bucket, key)
-
-
-def untar_file_from_s3(url: str, tmpdir: str, sha256sum: str):
+def untar_file_from_url(url: str, tmpdir: str, sha256sum: str):
     filename = "ostree.tar"
     filepath = os.path.join(tmpdir, filename)
 
-    # Grab file from s3
-    logger.info(f"Downloading object from s3: {url}")
-    s3 = boto3.client("s3")
-    bucket, key = parse_s3_url(url)
-    s3.download_file(bucket, key, filepath)
+    # Grab file from the url
+    logger.info(f"Downloading object from url: {url}")
+    urllib.request.urlretrieve(url, filepath)
```

Comment:
Just a side note, but we should also download the signature and verify it before importing. E.g. we can add a `commit-sig` to the message?

Reply:
We already check the sha256sum. The only thing I can think that the signature file would help us with is to make sure only signed content got into the repo (i.e. if we're worried about an attacker trying to send a message to import invalid content), but I can think of even better ways to do that too. One of them would be to verify the commit in the ostree repo after the file was untarred, but before it was imported.

Reply:
> One of them would be to verify the commit in the ostree repo after the file was untarred, but before it was imported.

Hmm, I thought that was already the case. We definitely should do this too if not. Ahh OK yup, we need to explicitly do `--gpgverify`.

> We already check the sha256sum. The only thing I can think that the signature file would help us with is to make sure only signed content got into the repo (i.e. if we're worried about an attacker trying to send a message to import invalid content)

One argument for requiring a signature on the tarball itself is to sanity check that [it's not malicious](https://en.wikipedia.org/wiki/Tar_(computing)#Tarbomb) before trying to extract it. Actually, even the docs for [TarFile.extractall()](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall) has a warning related to this.

So I think we should do both.

Reply:
so:

- remove checksum and apply signature checking for the tarball
- add commit checking in extracted ostree repo

?

Reply:
pushing that bit to a followup: https://github.com/coreos/fedora-coreos-releng-automation/issues/68